### PR TITLE
chore: refactor build config of all packages

### PR DESCRIPTION
**Pull request by targeral**
## Description

change build config and package.json entry field value.

Build products for node type projects:

1. `./dist/cjs`: `cjs`+ `es2019`.
2. `./dist/esm`: `esm` + `es2019`.
3. `./dist/types`:  `.d.ts` files.

Build products for universal type projects:

1. `./dist/cjs`: `cjs` + `es2019`.
2. `./dist/esm`: `esm` + `es5`.
3. `./dist/esm-node`: `esm` + `es2019`.
4. `./dist/types`: `.d.ts` files.

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [ ] New feature / Improvement
- [x] Refactoring
- [ ] Breaking change

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/cli/plugin-data-loader/src/cli/generate-client.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export const generateClient = ({
   let exportsCode = ``;
   const requestCreatorPath = path
     .join(__dirname, './create-request')
-    .replace('/node/cli/', '/treeshaking/cli/')
+    .replace('/cjs/cli/', '/esm-node/cli/')
```
[the remark this answers is not in the document]
Comment (yimingjfe) [reply packages/cli/plugin-data-loader/src/cli/generate-client.ts:23]: “esm-node” 这个应该要改为 “esm”

File: packages/toolkit/build-config/package.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "@modern-js/build-config",
+  "description": "A Progressive React Framework for modern web development.",
+  "homepage": "https://modernjs.dev",
+  "bugs": "https://github.com/modern-js-dev/modern.js/issues",
+  "repository": "modern-js-dev/modern.js",
+  "license": "MIT",
+  "version": "2.3.0",
```
Comment (chenjiahan) [inline packages/toolkit/build-config/package.json:8]: The latest version has been bumped to 2.4.0 in main branch

File: packages/toolkit/build-config/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  {
+    buildType: 'bundleless',
+    format: 'esm',
+    target: 'es5',
+    dts: false,
+    outDir: './dist/esm',
+  },
+  {
+    buildType: 'bundleless',
+    format: 'cjs',
```
Comment (chenjiahan) [inline packages/toolkit/build-config/src/index.ts:55]: Should be `esm` for `esm-node`

File: packages/cli/plugin-data-loader/src/cli/generate-client.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (targeral) [reply packages/cli/plugin-data-loader/src/cli/generate-client.ts:23]: 使用 es5 + esm 的产物是嘛？

**Review by chenjiahan — APPROVED** [review]

File: packages/cli/plugin-data-loader/src/cli/generate-client.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (yimingjfe) [reply packages/cli/plugin-data-loader/src/cli/generate-client.ts]: 对的，这个代码会跑在浏览器上的

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**